File: Example Project/Test/ViewController2.m
//
//  ViewController2.m
//  Test
//
//  Created by Prince Ugwuh on 8/10/16.
//  Copyright © 2016 TTT. All rights reserved.
//

#import "ViewController2.h"
#import "Test-Swift.h"
#import "TestCell.h"

@interface ViewController2 ()
@property(nonatomic, weak, nullable) IBOutlet UITableView *tableView;
@property(nonatomic, strong, nullable) MSTableViewDataSource *dataSource;
@end

@implementation ViewController2

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.tableView registerClass:[TestCell class] forCellReuseIdentifier:[TestCell identifier]];
    self.tableView.estimatedRowHeight = 54;
    self.tableView.rowHeight = 54;
    
    [self build];
}


- (void)build {
    
    NSMutableArray *items = @[].mutableCopy;
    
    for (NSInteger i = 0; i < 100; i++) {
        MSTableViewDataSourceCellItem *item = [[MSTableViewDataSourceCellItem alloc] init];
        item.cellIdentifier = [TestCell identifier];
        item.item = [NSString stringWithFormat:@"%ld", i];
        [items addObject:item];
    }
    
    self.dataSource = [[MSTableViewDataSource alloc] initWithCellItems:items];

    self.tableView.dataSource = self.dataSource;
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Example Projects/Example Project Swift/Example Project Swift/ViewController2.swift
//
//  SecondViewController.swift
//  Example Project Swift
//
//  Created by Prince Ugwuh on 10/12/16.
//  Copyright © 2016 MFX Studios. All rights reserved.
//

import UIKit

class ViewController2: UIViewController {
    @IBOutlet weak var tableView: UITableView!
    
    var dataSource: TableViewDataSource?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.tableView.register(TestCell.self, forCellReuseIdentifier: TestCell.identifier)
        self.tableView.estimatedRowHeight = 54
        self.tableView.rowHeight = 54
        
        build()
    }

    func build() {
        var items = [TableViewDataSourceCellItem]()
        for i in 0..<100 {
            let cellItem = TableViewDataSourceCellItem()
            cellItem.item = i
            cellItem.cellIdentifier = TestCell.identifier
            items.append(cellItem)
        }
        self.dataSource = TableViewDataSource(cellItems: items)
        self.tableView.dataSource = self.dataSource
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

